browser/module_test.go

```diff
@@ -45,3 +48,113 @@ func TestModuleNewDisabled(t *testing.T) {
 	}
 	assert.Panics(t, func() { New().NewModuleInstance(vu) })
 }
+
+// TestModuleMappings tests that all the methods of the API (api/) are
+// mapped to the module. This is to ensure that we don't forget to map
+// a new method to the module.
+func TestModuleMappings(t *testing.T) {
+	t.Parallel()
+
+	vu := &k6modulestest.VU{
+		RuntimeField: goja.New(),
+		InitEnvField: &k6common.InitEnvironment{
+			Registry: k6metrics.NewRegistry(),
+		},
+	}
+	rt := vu.Runtime()
+
+	type test struct {
+		apiObj any
+		mapp   func() mapping
+	}
+	mappers := map[string]test{
+		"browserType": {
+			apiObj: (*api.BrowserType)(nil),
+			mapp: func() mapping {
+				return mapBrowserType(rt, &chromium.BrowserType{})
+			},
+		},
+		"browser": {
+			apiObj: (*api.Browser)(nil),
+			mapp: func() mapping {
+				return mapBrowser(rt, &chromium.Browser{})
+			},
+		},
+		"browserContext": {
+			apiObj: (*api.BrowserContext)(nil),
+			mapp: func() mapping {
+				return mapBrowserContext(rt, &common.BrowserContext{})
+			},
+		},
+		"page": {
+			apiObj: (*api.Page)(nil),
+			mapp: func() mapping {
+				return mapPage(rt, &common.Page{})
+			},
+		},
+		"elementHandle": {
+			apiObj: (*api.ElementHandle)(nil),
+			mapp: func() mapping {
+				return mapElementHandle(rt, &common.ElementHandle{})
+			},
+		},
+		"frame": {
+			apiObj: (*api.Frame)(nil),
+			mapp: func() mapping {
+				return mapFrame(rt, &common.Frame{})
+			},
+		},
+		"mapRequest": {
+			apiObj: (*api.Request)(nil),
+			mapp: func() mapping {
+				return mapRequest(rt, &common.Request{})
+			},
+		},
+		"mapResponse": {
+			apiObj: (*api.Response)(nil),
+			mapp: func() mapping {
+				return mapResponse(rt, &common.Response{})
+			},
+		},
+	}
+
+	wildcards := wildcards()
+
+	for name, tt := range mappers {
+		tt := tt
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var (
+				typ    = reflect.TypeOf(tt.apiObj).Elem()
+				mapped = tt.mapp()
+			)
+			for i := 0; i < typ.NumMethod(); i++ {
+				method := typ.Method(i)
+				require.NotNil(t, method)
+
+				m := toFirstLetterLower(method.Name)
+				// change the method name if it is mapped to a wildcard
+				// method. these wildcard methods are not exist on our
+				// API. so we need to use the mapped method instead.
+				if wm, ok := isWildcard(wildcards, typ.Name(), m); ok {
+					m = wm
+				}
+				if _, ok := mapped[m]; !ok {
```

Comment:
Should we also ensure that `query` and `queryAll` aren't being mapped in the `goja.Object`?

Reply:
Good idea. Resolved in https://github.com/grafana/xk6-browser/pull/673/commits/220be028aae76e9a1e98489ed1da2cc6ccaa3b7b.